src/scripts/initialize-external-operator-identity.sh

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env bash
+# Copyright 2019, Oracle Corporation and/or its affiliates.  All rights reserved.
+# Licensed under the Universal Permissive License v 1.0 as shown at http://oss.oracle.com/licenses/upl.
+
+# do not turn on 'set -x' since it can print sensitive info, like secrets and private keys, to the operator log
+#set -x
+
+if [ "$#" != 0 ] ; then
+  1>&2 echo "Syntax: ${BASH_SOURCE[0]}"
+  exit 1
+fi
+
+EXTERNAL_CERT="externalOperatorCert"
+EXTERNAL_KEY="externalOperatorKey"
+OPERATOR_DIR="/operator"
+EXTERNAL_IDENTITY_DIR="${OPERATOR_DIR}/external-identity"
+NAMESPACE=`cat /var/run/secrets/kubernetes.io/serviceaccount/namespace`
+OPERATOR_CONFIG_DIR=${OPERATOR_DIR}/config
+OPERATOR_SECRETS_DIR=${OPERATOR_DIR}/secrets
+
+# the operator runtime expects the external operator cert and private key to be in these files:
+EXTERNAL_CERT_PEM="${EXTERNAL_IDENTITY_DIR}/${EXTERNAL_CERT}"
+EXTERNAL_KEY_PEM="${EXTERNAL_IDENTITY_DIR}/${EXTERNAL_KEY}"
+EXTERNAL_CERT_SECRET="${OPERATOR_CONFIG_DIR}/externalRestIdentitySecret"
+
+# the legacy helm install mount the ceritificate and private key in the following locations:
+LEGACY_CERT_BASE64_PEM=${OPERATOR_CONFIG_DIR}/${EXTERNAL_CERT}
+LEGACY_KEY_PEM=${OPERATOR_SECRETS_DIR}/${EXTERNAL_KEY}
+
+CACERT='/var/run/secrets/kubernetes.io/serviceaccount/ca.crt'
+TOKEN=`cat /var/run/secrets/kubernetes.io/serviceaccount/token`
+KUBERNETES_MASTER="https://kubernetes.default.svc"
+
+function cleanup {
+  if [[ $SUCCEEDED != "true" ]]; then
+    exit 1
+  fi
+}
+
+function getExternalIdentity {
+  SECRET_NAME=`cat ${EXTERNAL_CERT_SECRET}`
+  
+  curl -s \
```

Comment:
this seems odd - can't we just ready the file? where the secret is mounted?

Reply:
The user can choose the name of the secret so this is retrieving the name so that the secret can then be used from where it is mounted.